shell/browser/electron_download_manager_delegate.cc

```diff
@@ -130,6 +243,25 @@ void ElectronDownloadManagerDelegate::OnDownloadPathGenerated(
     const bool offscreen = !web_preferences || web_preferences->IsOffscreen();
     settings.force_detached = offscreen;
 
+#if BUILDFLAG(IS_WIN)
+    std::wstring extension = settings.default_path.FinalExtension();
+    if (!extension.empty() && settings.filters.empty()) {
+      std::vector<ui::FileFilterSpec> filter_spec = FormatFilterForExtensions(
+          {base::WideToUTF16(extension)}, {u""}, true, true);
+
+      std::string extension_spec_conv(
+          base::UTF16ToUTF8(filter_spec[0].extension_spec));
+      const std::vector<std::string> filter{extension_spec_conv};
+      std::string description_conv(
+          base::UTF16ToUTF8(filter_spec[0].description));
+
+      settings.filters.emplace_back(std::make_pair(description_conv, filter));
```

Comment:
The extension name generated by `FormatFilterForExtensions` has `.` prefix, which should be removed before passing to `file_dialog::ShowSaveDialog`.